src/autopas/particles/ParticleBase.h

```diff
@@ -325,4 +331,18 @@ class ParticleBase {
   friend void internal::markParticleAsDeleted(T &);
 };
 
+/**
+ * Stream operator for instances of ParticleBase class.
+ * This function enables passing ParticleBase objects to an ostream via `<<`
+ * @tparam Floating point type to be used for the SoAs.
+ * @param os
+ * @param D
+ * @return String representation.
+ */
+template <typename floatType, typename idType>
+std::ostream &operator<<(std::ostream &os, const ParticleBase<floatType, idType> &D) {
+  os << D.toString();
+  return os;
+}
```

Comment:
Same as with the ArrayUtils Stream operator you need to move the implementation of `ParticleBase::toString()` here. Otherwise manipulating the stream `os` has no effect on the string representation of `D`.
Also rename `D` to `particle`